src/confluent_kafka/src/Consumer.c

```diff
@@ -980,6 +980,33 @@ static PyObject *Consumer_poll (Handle *self, PyObject *args,
 }
 
 
+static PyObject *Consumer_memberid (Handle *self, PyObject *args,
+                                    PyObject *kwargs) {
+        char *memberid;
+        PyObject *memberidobj;
+        if (!self->rk) {
+                PyErr_SetString(PyExc_RuntimeError,
+                                "Consumer closed");
+                return NULL;
+        }
+
+        memberid = rd_kafka_memberid(self->rk);
+
+        if (!memberid)
+                Py_RETURN_NONE;
+
+        if (strlen(memberid) == 0) {
```

Comment:
nit: it is better to check if a string is empty by simply inspecting the first byte, which is O(1), rather than strlen() which is O(n). This doesn't matter at all here, of course, but it is good practice so mentioning it anyway:

```c
if (!*memberid) { 
```

Reply:
Thanks for the suggestion. I still think in Java terms I think. :)